src/tic-tac-toe.cpp

```diff
@@ -29,10 +35,110 @@ class Player {
     bool d_isCross;
 };
 
+class Field {
+public:
+    Field (int n) {
+        for (int i = 0; i < n; ++i) {
+            d_grid.push_back({});
+            for (int j = 0; j < n; ++j) {
+                d_grid.back().push_back(nullptr);
+            }
+        }
+    }
+
+    bool makeMove(const Player& player, int x, int y) {
+        if ((x < d_grid.size()) && (y < d_grid[x].size())) { // Please check the condition
+            if (d_grid[x][y] == nullptr) {
+                d_grid[x][y] = &player;
+                return true;
+            }
+        }
+        return false;
+    }
+
+    Status status(const Player& player) { // 0 -- win,  1 -- draw,  2 -- the game in progress
+        for (int i = 0; i < d_grid.size(); ++i) {
+            for (int j = 0; j < d_grid[i].size(); ++j) {
+                if ((checkWinDiagonals() == true) && (checkWinRows() == true) && (checkWinColumns() == true)) {
+                    return e_WIN;
+                }
+                else if (checkInProgress() == true) {
+                    return e_IN_PROGRESS;
+                }
+                else {
+                    return e_DRAW;
+                }
+            }
+        }
+    }
+    
+private:
+    vector<vector<const Player*>> d_grid;
+};
+
 void clearScreen() {
     system("cls");
 }
 
+// Functions for checking the win
+bool checkWinDiagonals() { // I don't know, what pass to the function
```

Comment:
If you have any questions you'd better leave your comments right in the review, like this one :)